apps/condo/domains/marketplace/schema/InvoiceContext.test.js

```diff
@@ -0,0 +1,297 @@
+                await createTestOrganizationEmployee(adminClient, o10n, client.user, role)
+
+                const objs = await InvoiceContext.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs.length).toBeGreaterThanOrEqual(1)
+                expect(objs).toEqual(expect.arrayContaining([
+                    expect.objectContaining({
+                        id: objCreated.id,
+                    }),
+                ]))
+            })
+
+            test('staff without permission can\'t', async () => {
+                const [o10n] = await createTestOrganization(adminClient)
+                await createTestInvoiceContext(adminClient, o10n)
+
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+                const [role] = await createTestOrganizationEmployeeRole(adminClient, o10n, { canReadInvoiceContexts: false })
+                await createTestOrganizationEmployee(adminClient, o10n, client.user, role)
+
+                const objs = await InvoiceContext.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs).toHaveLength(0)
+            })
+
+            test('each staff user can read only his organization context', async () => {
+                const [o10n1] = await createTestOrganization(adminClient)
+                const [o10n2] = await createTestOrganization(adminClient)
+
+                const [objCreated1] = await createTestInvoiceContext(adminClient, o10n1)
+                const [objCreated2] = await createTestInvoiceContext(adminClient, o10n2)
+
+                const client1 = await makeClientWithNewRegisteredAndLoggedInUser()
+                const client2 = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                const [role1] = await createTestOrganizationEmployeeRole(adminClient, o10n1, { canReadInvoiceContexts: true })
+                await createTestOrganizationEmployee(adminClient, o10n1, client1.user, role1)
+
+                const [role2] = await createTestOrganizationEmployeeRole(adminClient, o10n2, { canReadInvoiceContexts: true })
+                await createTestOrganizationEmployee(adminClient, o10n2, client2.user, role2)
+
+                const objs1 = await InvoiceContext.getAll(client1, {}, { sortBy: ['updatedAt_DESC'] })
+                const objs2 = await InvoiceContext.getAll(client2, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs1).toEqual([expect.objectContaining({ id: objCreated1.id })])
+                expect(objs2).toEqual([expect.objectContaining({ id: objCreated2.id })])
+            })
+
+            test('anonymous can\'t', async () => {
+                const [o10n] = await createTestOrganization(adminClient)
+                await createTestInvoiceContext(adminClient, o10n)
+
+                const client = await makeClient()
+                await expectToThrowAuthenticationErrorToObjects(async () => {
```

Comment:
Accessories are checked before searching the database, so you don’t need to create entities to check for anonymus

Reply:
fixed